### PR TITLE
Review and Merge the pull request on APIMANAGER-2722

### PR DESCRIPTION
**Pull request by tharindu1st**
https://wso2.org/jira/browse/APIMANAGER-2722

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -2897,9 +2897,10 @@ public static boolean jsFunction_addDocumentation(Context cx, Scriptable thisObj
             if (fileHostObject != null && fileHostObject.getJavaScriptFile().getLength() != 0) {
             	String contentType = (String) args[10];
                 Icon icon = new Icon(fileHostObject.getInputStream(), contentType);
-                
-                String filePath = APIUtil.getDocumentationFilePath(apiId, fileHostObject.getName());
                 String fname = fileHostObject.getName();
+                int index = fname.lastIndexOf("//");
+                fname = fname.substring(index+1);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: Have a spaces before and after +

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -2835,7 +2835,7 @@ private static void addAPIObj(SubscribedAPI subscribedAPI, NativeArray apisArray
             apiObj.put("hasMultipleEndpoints", apiObj, String.valueOf(api.getSandboxUrl() != null));
             apisArray.put(apisArray.getIds().length, apisArray, apiObj);
         } catch (APIManagementException e) {
-            handleException("Error while obtaining application metadata", e);
+            log.error("Error while obtaining application metadata", e);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Any reason for just log the exception as it was previously logged and throw the exception with handleException?

Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: according to the above mentioned jira if registry artifact of api is removed frrom the management console and go to access store will occur this exception as  this exception throws api store couldn't show the apis if registry corruption happen

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java (hunk 1 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java]: code block not available in latest code

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 2 above)
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Please put a comment explaining why we're not throwing the exception.

Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: fixed 
https://github.com/wso2/carbon-apimgt/pull/792

**Pull request merged**